### ISSUE TITLE
:nxdomain when calling AWS.CloudFront.list_distributions

### ISSUE THREAD

**Issue by grant-zukowski-xumo** [aws-beam/aws-elixir#186]
AWS CLI call works when calling locally.  It is just when I make this call using the AWS library that I get back 
`{:error, %Mint.TransportError{reason: :nxdomain}}`

I tried changing clients and that didn't change anything.  My default region is "us-east-1"

This is on version 0.14.1
Confirmed the call still works on 0.13.0

I upgraded so I could use the new MP v2 modules.

One of my colleagues figured out that it has to do with the endpoint being set to nil on all clients:

```
:arrow_up: is not exactly true. Endpoint in the client options is nil. When it's nil the API library will build the endpoint based on AWS services. For global services it is adding region to the endpoint which doesn't work for services like CloudFront. We are overriding the endpoint to make it work.
```

![image](https://github.com/aws-beam/aws-elixir/assets/146299357/8d75d0fd-1501-4153-adf7-c70c4e3b15e6)

Most likely the commit that caused the problem:

https://github.com/aws-beam/aws-elixir/commit/1f87f00f7efabbd577c61045e57fa6478cfd756b

**Comment by onno-vos-dev** [comment]
Hi @grant-zukowski-xumo Sorry for missing this issue. This will be fixed by: https://github.com/aws-beam/aws-elixir/pull/191

Will ping here once the release is out 👍

**Comment by onno-vos-dev** [comment]
@grant-zukowski-xumo Release is out: https://hex.pm/packages/aws/1.0.1

**Comment by grant-zukowski-xumo** [comment]
@onno-vos-dev thank you very much.  I can't seem to pull in the new version.  


mix file:

`{:aws, "~> 1.0.1"},`


```
% mix deps.get                                                                                                                           
Resolving Hex dependencies...
Resolution completed in 0.023s
Because your app depends on aws empty which doesn't match any versions, version solving failed.
```

**Comment by onno-vos-dev** [comment]
Hmm @grant-zukowski-xumo can you try again and if it still doesn't work, show your deps? 🤔 

I just did `mix new foo` and modified the `mix.exs` to have:
```elixir
defp deps do
    [ {:aws, "~> 1.0.1"}
    ]
  end
```

Which results in:
```
mix deps.get
Resolving Hex dependencies...
Resolution completed in 0.035s
New:
  aws 1.0.1
  aws_signature 0.3.2
  jason 1.4.1
* Getting aws (Hex package)
* Getting aws_signature (Hex package)
* Getting jason (Hex package)
```

and hence seems to work fine 🤔

**Comment by grant-zukowski-xumo** [comment]
@onno-vos-dev confirmed, I was able to get the latest.  

Random question, I've always updated each dependency one at at time then called `mix deps.get`.  I've recently learned of the style of doing `mix deps.update --all`.  Just wondering if you have a preference.  

Confirmed that I can get the new version, lots changed with the new codegen I think, but I'll confirm here that everything works once I satisfy dialyzer.
